Center InternalDatePicker Clear button

leojalfred opened the pull request:
Addresses https://github.com/twentyhq/twenty/issues/3249 by altering the `InternalDatePicker` Clear button to take the full width. Note that `.hoverable-buttons` is now conditionally rendered to prevent issues when centering the items in the button.

<img width="348" alt="Screenshot 2024-01-05 at 12 07 48 PM" src="https://github.com/twentyhq/twenty/assets/12378049/37912d8f-d570-4abd-ada3-722c27f0030e">
<img width="347" alt="Screenshot 2024-01-05 at 12 08 00 PM" src="https://github.com/twentyhq/twenty/assets/12378049/a256aef9-eac6-48d4-a51d-ecfbb44db1f0">

This is my first open source contribution, please let me know if I can do anything to streamline my contribution.

charlesBochet commented on packages/twenty-front/src/modules/ui/input/components/internal/date/components/InternalDatePicker.tsx:
```diff
@@ -231,14 +231,17 @@ const StyledButtonContainer = styled.div`
     padding: 0 ${({ theme }) => theme.spacing(2)};
-    width: fit-content;
+
+    & > div {
```
We avoid styling children through CSS.
Instead you can transform the div itself to a Styled component

lucasbordeau commented:
Hi @leojalfred, it seems that you're both working on the same issue with @abdul-irfan-k, see other PR : https://github.com/twentyhq/twenty/pull/3273

Would you be ok to work together on this ? 

We need this to be pixel perfect like in the figma.

leojalfred commented:
@lucasbordeau I'd be happy to. I can push up necessary changes tomorrow morning. Do you have a specific example in Figma that you want me to reference? I'm looking at the calendar components linked on the repo's README and I don't see a mockup for this.

charlesBochet commented:
@leojalfred Thank you :)
The figma is located here: https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=2709-52753&mode=design&t=bTCrlh4rIYbzufiM-0

leojalfred commented:
@charlesBochet I saw that but that doesn't really have much to do with the specifics of this ticket. Also the existing implementation doesn't match the design in Figma, particularly with respect to the padding. I can do my best when it comes to the padding, but it makes more sense to have the spacing for the Clear button line up with the rest of the spacing in existing implementation rather than the design in Figma.

leojalfred commented:
Also the div that you pointed out in the comment was already a styled component (`StyledMenuItemLeftContent`). The reason I used the div in styling is because it avoided the need for prop drilling. I think in order to not use children in the styling I'll need to pass a centering boolean down the chain to that component and use it to add `justify-content: center`.

I've got a question about using children in styling. It seems like children are used in styling, as in targeting `.menu-item` under the `StyledButtonContainer` styles. Should this be refactored as well? Or is there some rule that determines when it's appropriate to use children in styling and when it's not?

leojalfred commented:
@charlesBochet it should be good to go. I changed the justify-content: center that was originally styled on the div to be styled via props in `StyledMenuItemLeftContent`. Per the other PR I changed `StyledButtonContainer` to extend `MenuItem`. Per the other PR I also changed the `StyledButtonContainer` spacing to be the same vertically and horizontally, but to my eye it looks worse.

lucasbordeau reviewed (changes requested):
Good for the design.

Could you just create the button from scratch and not use the MenuItem component ? 

Because MenuItem is used everywhere in the code and refactoring it to handle centered text is out of the scope of this issue. 

It doesn't play well with isCentered set to true and right icon buttons.

<img width="629" alt="image" src="https://github.com/twentyhq/twenty/assets/26528466/555f3d14-26fd-449d-a9f8-56538ded4727">
